packages/rum/src/domain/record/observers.spec.ts

```diff
@@ -154,3 +154,142 @@ describe('initFrustrationObserver', () => {
+        expect(styleSheetRule.adds?.pop()?.index).toEqual(undefined)
+      })
+
+      it('should capture CSSStyleRule insertion when index is provided', () => {
+        // Given
+        const index = 0
+        // When
+        stopStyleSheetObserver = initStyleSheetObserver(styleSheetCallbackSpy)
+        styleSheet.insertRule(styleRule, index)
+        // Then
+        const styleSheetRule = styleSheetCallbackSpy.calls.first().args[0]
+        expect(styleSheetRule.id).toBeDefined()
+        expect(styleSheetRule.removes).toBeUndefined()
+        expect(styleSheetRule.adds?.length).toEqual(1)
+        expect(styleSheetRule.adds?.pop()?.index).toEqual(index)
+      })
+    })
+
+    describe('when removing rules from stylesheet', () => {
+      it('should capture CSSStyleRule removal with the correct index', () => {
+        styleSheet.insertRule(styleRule)
+        // Given
+        const index = 0
+        // When
+        stopStyleSheetObserver = initStyleSheetObserver(styleSheetCallbackSpy)
+        styleSheet.deleteRule(index)
+        // Then
+        const styleSheetRule = styleSheetCallbackSpy.calls.first().args[0]
+        expect(styleSheetRule.id).toBeDefined()
+        expect(styleSheetRule.adds).toBeUndefined()
+        expect(styleSheetRule.removes?.length).toEqual(1)
+        expect(styleSheetRule.removes?.pop()).toEqual({ index })
+      })
+    })
+  })
+
+  describe('observing CSSGroupingRules inside a CSSStyleSheet', () => {
+    describe('when inserting CSSRules inside a CSSGroupingRule', () => {
+      it('should capture CSSRule with the correct path when no index is provided', () => {
+        styleSheet.insertRule('@media cond-2 { @media cond-1 { .nest-1 { color: #ccc } } }')
+        styleSheet.insertRule('.main {opacity: 0}')
+        // Given
+        const groupingRule = (styleSheet.cssRules[1] as CSSGroupingRule).cssRules[0] as CSSGroupingRule
+        // When
+        stopStyleSheetObserver = initStyleSheetObserver(styleSheetCallbackSpy)
+        groupingRule.insertRule(styleRule, 1)
+        // Then
+        const styleSheetRule = styleSheetCallbackSpy.calls.first().args[0]
+        expect(styleSheetRule.id).toBeDefined()
+        expect(styleSheetRule.removes).toBeUndefined()
+        expect(styleSheetRule.adds?.length).toEqual(1)
+        expect(styleSheetRule.adds?.pop()?.index).toEqual([1, 0, 1])
+      })
+
+      it('should not create record when inserting into a detached CSSGroupingRule', () => {
+        styleSheet.insertRule('@media cond-2 { @media cond-1 { .nest-1 { color: #ccc } } }')
+        // Given
+        const parentRule = styleSheet.cssRules[0] as CSSGroupingRule
+        const groupingRule = parentRule.cssRules[0] as CSSGroupingRule
+        parentRule.deleteRule(0)
+        // When
+        stopStyleSheetObserver = initStyleSheetObserver(styleSheetCallbackSpy)
+        groupingRule.insertRule(styleRule, 0)
+        // Then
+        expect(styleSheetCallbackSpy.calls.all().length).toEqual(0)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
        expect(styleSheetCallbackSpy).not.toHaveBeenCalled()
```